db/fixtures/ae_datastore/Deployment/System/StateMachine/Deployment.class/__methods__/add_provider.rb

```diff
@@ -0,0 +1,66 @@
+require 'net/ssh'
+require 'rest_client'
+
+def provider_token
+  token = ""
+  Net::SSH.start($evm.root['deployment_master'], $evm.root['user'], :paranoid => false, :forward_agent => true,
+                 :key_data => $evm.root['private_key']) do |ssh|
+    cmd = "oc get -n management-infra sa/management-admin --template='{{range .secrets}}{{printf " + '"%s\n"' \
+          " .name}}{{end}}' | grep token"
+    token_key = ssh.exec!(cmd)
+    token_key.delete!("\n")
+    cmd = "oc get -n management-infra secrets #{token_key} --template='{{.data.token}}' | base64 -d"
+    token = ssh.exec!(cmd)
+  end
+  token
+end
+
+def add_provider
+  $evm.root['state'] = "add_provider"
+  $evm.log(:info, "**************** add_provider ****************")
+  deployment_master = $evm.root['deployment_master']
+  token = provider_token
+
+  url = $evm.root['manageiq_url']
+  query = "/api/providers"
+  post_params = {
+    :credentials => {
+      :auth_type => "bearer",
+      :auth_key  => token
+    },
+    :type        => "ManageIQ::Providers::Openshift::ContainerManager",
+    :name        => $evm.root['provider_name'],
+    :port        => "8443",
+    :hostname    => deployment_master,
+    :ipaddress   => deployment_master
+  }.to_json
+
+  begin
+    rest_return = RestClient::Request.execute(
+      :method     => :post,
+      :url        => url + query,
+      :user       => $evm.root['manageiq_user'],
+      :password   => $evm.root['manageiq_password'],
+      :headers    => {:accept => :json},
+      :payload    => post_params,
+      :verify_ssl => false)
+
+    result = JSON.parse(rest_return.body)
+    $evm.root['provider_id'] = result["results"].first["id"]
```

Comment:
@alongoldboim 
Can we validate the 'result' hash to ensure that it has a 'results' and 'id' keys before we use it.